### ISSUE TITLE
Focus-trap start marker doesn't work when focus is redirected to an internal element

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#6029]
If `cdk-focus-start` / `-end` is applied to, e.g., `<md-radio-button>`, the focus trap will try to focus the actual `md-radio-button` element. This is problematic because the radio button actually defers focus/keyboard interaction to an internal `<input type="radio">` element.

**Comment by mmalerba** [comment]
@jelbourn is this an issue with the focus trap or the radio button? IMO it's a missing part of the radio button API, it should detect if someone is trying to focus it and redirect that focus appropriately.

**Comment by jelbourn** [comment]
The radio button's `focus` method focuses the right thing, but it doesn't do anything if you `.focus()` the native element. It's probably okay to set up a `focus` listener on the host and redirect, though.
